cli/deno_dir.rs

```diff
@@ -280,51 +280,13 @@ impl DenoDir {
       }
     }
 
-    // We're dealing with remote file, first try local cache
-    if use_disk_cache {
-      match self.fetch_cached_remote_source(&module_url, None) {
-        Ok(Some(source_file)) => {
-          return Either::A(futures::future::ok(source_file));
-        }
-        Ok(None) => {
-          // there's no cached version
-        }
-        Err(err) => {
-          return Either::A(futures::future::err(err));
-        }
-      }
-    }
-
-    // If remote file wasn't found check if we can fetch it
-    if no_remote_fetch {
-      // We can't fetch remote file - bail out
-      return Either::A(futures::future::err(
-        std::io::Error::new(
-          std::io::ErrorKind::NotFound,
-          format!(
-            "cannot find remote file '{}' in cache",
-            module_url.to_string()
-          ),
-        ).into(),
-      ));
-    }
-
     // Fetch remote file and cache on-disk for subsequent access
-    // not cached/local, try remote.
-    let module_url_ = module_url.clone();
-    Either::B(self
-        .fetch_remote_source_async(&module_url)
-        // TODO: cache fetched remote source here - `fetch_remote_source` should only fetch with
-        // redirects, nothing more
-        .and_then(move |maybe_remote_source| match maybe_remote_source {
-          Some(output) => Ok(output),
-          None => Err(
-            std::io::Error::new(
-              std::io::ErrorKind::NotFound,
-              format!("cannot find remote file '{}'", module_url_.to_string()),
-            ).into(),
-          ),
-        }))
+    Either::B(self.fetch_remote_source_async(
+      &module_url,
+      use_disk_cache,
+      no_remote_fetch,
+      10,
```

Comment:
I think we should allow for 10-20 redirects CC @ry

Reply:
that's fine